support/db/metrics.go

```
@@ -356,6 +369,55 @@ func getQueryType(ctx context.Context, query squirrel.Sqlizer) QueryType {
 	return UndefinedQueryType
 }
 
+func (s *SessionWithMetrics) handleErrorEvent(dbErr error, callerContext context.Context) {
+	if dbErr == nil || s.NoRows(dbErr) {
+		return
+	}
+
+	abendOrigin := "libpq"
+	abendType := "error"
+	abendCondition := ""
+	var abendDbErrorCode pq.ErrorCode
+
+	if err, ok := dbErr.(*pq.Error); ok {
```

we should use https://pkg.go.dev/errors#As because that will handle cases where the db error is hidden under several layers of wrapping

---

done, https://github.com/stellar/go/pull/5225/commits/26c3316ab8f1a1acdbc17706821139f22ad8820b